openssl/params.go

```diff
@@ -99,3 +101,15 @@ func (pb *paramsBuilder) addBigNumber(key *C.char, v []byte) error {
 	pb.add(key, C.GO_OSSL_PARAM_UNSIGNED_INTEGER, unsafe.Pointer(cbytes), C.size_t(len(v)))
 	return nil
 }
+
+// addBN adds a parameter of type OSSL_PARAM_UNSIGNED_INTEGER.
+func (pb *paramsBuilder) addBN(key *C.char, v C.GO_BIGNUM_PTR) error {
```

Comment:
Now we have `addBigNumber` and `addBN`, which looks a bit misleading; perhaps we could rename the former?

By the way, in #55 I use `OSSL_PARAM_BLD_*` in the C level, and `OSSL_PARAM_BLD_push_BN` seems to take care of the endianness conversion.

Reply:
Thanks for the suggestion @ueno. I've realized that `paramsBuilder` was causing more harm than good, so I've removed it and changed all usages to the `OSSL_PARAM_BLD` interface.